models/users.js

```diff
@@ -0,0 +1,216 @@
+'use strict';
+
+var Db        = require('../config/database');
+var Session   = require('./sessions');
+var validator = require('validator');
+var mongoose  = require('mongoose');
+
+// Create the schema for a user
+var UserSchema = mongoose.Schema({
+  username      : { type: String, required: true },
+  usernameLower : { type: String, required: true },
+  email         : { type: String, required: true },
+  password      : { type: String, required: true }
+});
+
+// Creates a collection named users in MongoDB
+var UserMongoModel = Db.model('users', UserSchema);
+
+// Takes username, password, email and saves a new user to MongoDB
+function createUser(username, password, email, callback) {
+  var newUser = new UserMongoModel({
```

Comment:
We should define newUser after checking if the username and email are not taken.